### DISCUSSION TITLE
Use Livewire php variables in javascript

### DISCUSSION THREAD

**Discussion by MotionPhix** [livewire/livewire#6151] [1. Help] [answered]
I have a scenerio where I am listening to an event and change an Id of the newly created company from the event. That works, however, I am unable to figure out how to use the value I get from the event in Javascript like below:

```
#[On('update-selected-company')]
  public function updateCompany($id)
  {
    $this->js(<<<'JS'
      setTimeout(() => {
        $wire.company_id = {{ $id }};
      }, 250);
    JS);
  }
```

The above code does not work. How can I grab that Id and use it properly?

**Comment by wizzymore** [comment]
You can try this:
```
#[On('update-selected-company')]
  public function updateCompany($id)
  {
    $this->js(<<<JS
      setTimeout(() => {
        $wire.company_id = {$id};
      }, 250);
    JS);
  }
```

You need to make it a HEREDOC first and then you can pass variables using the complex or simple syntax

**Reply by MotionPhix** [reply]
The above doesn't still work. If I do:

`console.log({$id});` inside the setTimeout function, I get an object in the console which doesn't contain any values, just a weird function. Really wish there was a way to get the values from the component into these JS sugarcoats. It would really help if these were also thoroughly documented, tackling situations like this. If I do:

```
#[On('update-selected-company')]
  public function updateCompany($id)
  {
    $this->dispatch('update-id', id: $id);
  }
```
and then listen for this event from the frontend, like:

```
<script>
  document.addEventListener('livewire:init', function () {
    Livewire.on('update-id', function (data) {
      let components = Livewire.getByName('contacts.create')

      setTimeout(() => {
        components[0].set('company_id', data.id);
      }, 250);
    });
  });
</script>
```
I get the intended result, but I just think it's a hack it doesn't sit well with me. I wanted it in the component, not to write anything in the frontend

**Reply by wizzymore** [reply]
Have you removed the quotes from here: `<<<JS`?

**Reply by wizzymore** [reply]
In your code you are using a newdoc, you can't pass variables to newdocs, so switch it to a heredoc by removing the quotes `<<<'JS'` -> `<<<JS`

**Reply by MotionPhix** [reply]
Would you mind showing it in a code? I can't get what you are trying to say. If I do:

```
#[On('update-selected-company')]
  public function updateCompany($id)
  {
    $this->js(<<<JS
      setTimeout(() => {
        $this->company_id = $id;
      }, 50);
   JS );
  }
```
It gives me errors saying 

```
Alpine Expression Error: Malformed arrow function parameter list

Expression: "  setTimeout(fn () => {
     = 63;
  }, 50);"

```

If I do:

```
#[On('update-selected-company')]
  public function updateCompany($id)
  {
    $this->js(<<<JS
      setTimeout(() => {
        $wire.company_id = $id;
      }, 50);
   JS );
  }
```
It's saying $wire is not defined. So how do I put it exactly to make it work?

**Reply by joshhanley** [reply]
@MotionPhix try this

```php
#[On('update-selected-company')]
  public function updateCompany($id)
  {
    $this->js(<<<JS
      setTimeout(() => {
        \$wire.company_id = $id;
      }, 50);
   JS );
  }
  ```
  
  As `$wire` is a javascript helper, you need to escape the `\$` otherwise PHP will try and interpret it.
  
  Hope this helps!

**Comment by MotionPhix** [comment, marked as answer]
Ok, finally I got it to work: here is how I implemented it.

 ```
 #[On('update-selected-company')]
  public function updateCompany($id)
  {
    $this->js(<<<JS
      let component = Livewire.getByName('contacts.create')[0]

      setTimeout(() => {
        component.set('company_id', $id);
      }, 50);
    JS);
  }
```

Thank you so much for pointing out that I should use `heredoc` @wizzymore
